src/java/org/apache/cassandra/cql3/statements/DeleteStatement.java

```diff
@@ -177,6 +175,8 @@ protected ModificationStatement prepareInternal(TableMetadata metadata,
 
             if (stmt.hasConditions() && !restrictions.hasAllPKColumnsRestrictedByEqualities())
             {
+                checkFalse(stmt.isVirtual(), "DELETE statements must restrict all PRIMARY KEY columns with equality relations");
```

Comment:
This check is to ensure a proper message (without mentioning static columns) is shown when not all PKs are specified with a condition.